src/ResourceType.php

```diff
@@ -40,6 +40,11 @@ public function getClass()
         return $this->configuration['class'];
     }
 
+    public function getQuery()
+    {
+        return $this->configuration['query'] ?? $this->getClass()::query();
```

Comment:
I think I would rather see something like this. This would also work if 'query' is not set in the configuration.
~~~.php
use Illuminate\Support\Arr;
// [...]
return Arr::get($this->configuration, 'query', $this->getClass()::query());
~~~